src/prefect/new_task_engine.py

```diff
@@ -435,11 +435,10 @@ def start(
         """
         Enters a client context and creates a task run if needed.
         """
-
-        with get_client(sync_client=True) as client:
-            self._client = client
-            self._is_started = True
-            with hydrated_context(self.context, client=client):
+        with hydrated_context(self.context):
+            with get_client(sync_client=True) as client:
+                self._client = client
+                self._is_started = True
```

Comment:
I had to change this order to ensure the client gets the correct API URL from the current settings.